src/supertux/tile_set_parser.cpp

```diff
@@ -70,10 +81,12 @@ TileSetParser::parse()
     else if (iter.get_key() == "tiles")
     {
       ReaderMapping tiles_mapping = iter.as_mapping();
-      parse_tiles(tiles_mapping);
+      parse_tiles(tiles_mapping, start, end, offset);
     }
     else if (iter.get_key() == "autotileset")
     {
+      /* ignore autotiles for imported tilesets unless there's no limit and no offset */
+      if (start || end || offset) continue;
```

Comment:
It would be possible to load autotilesets by giving the loader the offset and handling it inside the parser.

That being said, how are imported tilesets handled in the editor? Is it a manual-only feature?

Reply:
The offset isn't the problem, the problem is that you can only import a part of the tileset (for example, in a hypothetical situation, you may only want to import tiles 100-500 from a tileset, but its autotiles try to use tiles 501, 502 and 424242 with tile 500. What now?) If you import a reasonable range from tiles.strf without an offset, you can easily use the autotiles by just putting `(autotileset (source "autotiles.satc"))` in your strf.

As for the editor, there's currently no way to define your own tilesets from within the game, right? So imported tilesets are a feature you can use when defining your custom strf. `(import-tileset)` is just a more convenient way of using existing tiles (than manually copying them from said tileset and possibly remapping their ids), and it also guarantees that if something like a path to an image changes in the imported tileset, yours won't need to be changed to accommodate for that changed file.

Reply:
It isn't possible to *write* a custom .strf file from withing the editor, but it is very possible to *assign* a .strf file in the editor, hence the question: how will the editor handle levels that use multiple tileset files?

Reply:
The editor doesn't even know the tileset uses imported tiles, it works as it should. Quick demonstration:

https://user-images.githubusercontent.com/14074789/147425995-cc76f80f-587f-40bd-893c-34dc1fb3123d.mp4

The `test_tiles.strf` tileset imports tiles 0-500 from `tiles.strf`, assigns them ids 1000-1500, and defines 4 new tiles with ids 1-4. Then there are 3 tilegroups defined, one with some imported tiles, one with some new tiles, and one with a mix of both. Everything else is correctly put into the unassigned "Others" tilegroup, with the correct ids.